src/main/java/mate/academy/service/impl/AuthenticationServise.java

```diff
@@ -0,0 +1,43 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServise implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isEmpty()) {
+            throw new AuthenticationException("Can't find user by email");
+        }
+        User userFromDB = userFromDbOptional.get();
+        String hashedPassword = HashUtil.hashPassword(password, userFromDB.getSalt());
+        if (userFromDB.getPassword().equals(hashedPassword)) {
+            return userFromDB;
+        }
```

Comment:
Redo the code so that there is only one if and one throw

Reply:
Done